docs/source/extension/extension_points.rst

```diff
@@ -457,16 +457,174 @@ The recommended ranges for this rank are:
 Main Menu
 ---------
 
-There are three main ways to extend JupyterLab's main menu.
+There are two ways to extend  JupyterLab's main menu.
+
+1. Using the settings - this is the preferred way as they are configurable by the user.
+2. Using the API - this is for advanced cases like dynamic menu or semantic items.
+
+Settings-defined menu
+^^^^^^^^^^^^^^^^^^^^^
+
+JupyterLab provides integration with its settings system for menu definitions.
+Your extension can provide a settings schema with a ``jupyter.lab.menus`` key,
+declaring default menus.
+
+To add a new menu with your extension command:
+
+.. code:: json
+
+    {
+      "jupyter.lab.menus": {
+      "main": [
+        {
+          "id": "jp-mainmenu-myextension",
+          "items": [
+            {
+              "command": "my-command",
+              "rank": 500
+            }
+          ],
```

Comment:
How is the label of the new menu provided by extension determined? Is extra code in the extension needed to define the menu?

Reply:
Thanks for the review I updated the documentation to clarify that point.

Reply:
Thanks, the update certainly helps! I actually meant the menu label, not the items labels, like "File", "Edit", or "Git" in git extension so label for `jp-mainmenu-myextension` (which I understand is on the same level as `jp-mainmenu-file`, `jp-mainmenu-file`, etc) and I still do not know how to set the label for that.

Reply:
For menus (and submenus), the label needs to be set explicitly with the ``label`` property.

> I added that in the documentation as well.

Reply:
I see, sorry I missed it. Should there be a code example for setting it explicitly? again, sorry if it is already there I might be just looking in a wrong spot.

Reply:
That will be a good addition - I'll update the PR.